src/App.ts

```diff
@@ -366,19 +368,22 @@ export default class App {
       };
     }
 
-    // Initialize receiver
+    /* --------------------- Initialize receiver ---------------------- */
     if (receiver !== undefined) {
-      // Custom receiver
+      // Custom receiver supplied
       if (this.socketMode) {
-        throw new AppInitializationError('receiver cannot be passed when socketMode is set to true');
+        // socketMode = true should result in SocketModeReceiver being used as receiver
+        // TODO: Add case for when socketMode = true and receiver = SocketModeReceiver
+        // this should not result in error
+        throw new AppInitializationError('You cannot supply a custom receiver when socketMode is set to true.');
       }
       this.receiver = receiver;
-    } else if (this.socketMode) {
+    } else if (this.socketMode === true) {
```

Comment:
I think either is fine. Do you have some situation that we can improve by changing this way in mind?

Reply:
@seratch - Yes, I was trying to make it easier to read. This page contains lots of different conditions and my main goal is to improve readability. Since `socketMode` variable name doesn't make it clear that it's a boolean (like the way `usingOauth` is evidently a boolean).

Reply:
Thanks for your reply. While I can understand your point on readability, the code like `something === true` makes me think of the pattern to avoid accepting truthy values (say, `1` etc.). That's why I asked the above question.

I don't have any strong opinion here and am fine to change the code. But in the case, L374 is inconsistent with it.

Reply:
Yes, that's a great point that in addition to added readability, adding more specificity to the condition helps us avoid ES values which evaluate as truthy like `1`. I've gone ahead and made the same change to L374. Thanks!